### PR TITLE
[Supplier] feat: add supplier un/stake events

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Adds the following events:
- `EventSupplierStaked`
- `EventSupplierUnbondingBegin`
- `EventSupplierUnbondingEnd`

## Depends on

- #815 

## Dependents

- #826

## Issue

Results in more robust supplier.feature steps.

- #799

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: proto/poktroll/supplier/event.proto [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import "poktroll/shared/supplier.proto";
+
+// EventSupplierStaked is emitted with the commitment of the supplier is stake message.
+message EventSupplierStaked {
+  poktroll.shared.Supplier supplier = 1;
+}
+
+// EventSupplierUnstaked is emitted with the commitment of the supplier is unstake message.
+message EventSupplierUnbondingBegin {
+  poktroll.shared.Supplier supplier = 1;
```
Comment (red-0ne) [inline proto/poktroll/supplier/event.proto:19]: Would be interesting to include the unbonding height here (If it's not too difficult to calculate it).

File: proto/poktroll/supplier/event.proto [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import "gogoproto/gogo.proto";
+
+import "poktroll/shared/supplier.proto";
+
+// EventSupplierStaked is emitted with the commitment of the supplier is stake message.
+message EventSupplierStaked {
+  poktroll.shared.Supplier supplier = 1;
+}
+
+// EventSupplierUnstaked is emitted with the commitment of the supplier is unstake message.
```
Comment (red-0ne) [inline proto/poktroll/supplier/event.proto:17]:
```suggestion
// EventSupplierUnstaked is emitted with the commitment of the supplier unstake message.
```

File: proto/poktroll/supplier/event.proto [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+option go_package = "github.com/pokt-network/poktroll/x/supplier/types";
+option (gogoproto.stable_marshaler_all) = true;
+
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+import "poktroll/shared/supplier.proto";
+
+// EventSupplierStaked is emitted with the commitment of the supplier is stake message.
```
Comment (red-0ne) [inline proto/poktroll/supplier/event.proto:12]:
```suggestion
// EventSupplierStaked is emitted with the commitment of the supplier stake message.
```

**Review by red-0ne — APPROVED** [review]
Feels so good to have these events :smile: 

Preemptively approving but please consider the request about adding the unbonding height in the supplier unstaked event.

**Review by Olshansk — APPROVED** [review]

**Pull request merged**